src/js_tests/styledelements/GUIBuilderSpec.js

```diff
@@ -307,6 +306,36 @@
             var outerHTML = result.elements[0].outerHTML.replace(' xmlns="http://www.w3.org/1999/xhtml"', '');
             expect(outerHTML).toBe('<div><div class="se-btn" tabindex="0"><span>Save</span></div></div>');
         });
+
+        it("should allow to nest template elements with function context and json option", function () {
```

Comment:
I tried to construct a complex test case with the different possibility which I found, and mimicking the use case of the user menu

Reply:
more than enough.